k8s/apis/fabric/v1/target_webhook.go

```diff
@@ -81,6 +82,25 @@ func (r *Target) Default() {
 	if r.Spec.ReconciliationPolicy != nil && r.Spec.ReconciliationPolicy.State == "" {
 		r.Spec.ReconciliationPolicy.State = v1.ReconciliationPolicy_Active
 	}
+
+	if r.Spec.RootResource != "" {
+		var targetContainer TargetContainer
+		err := myTargetClient.Get(context.Background(), client.ObjectKey{Name: r.Spec.RootResource, Namespace: r.Namespace}, &targetContainer)
```

Comment:
Do you add indexfield on root resource?

Reply:
Checked with @LeiQL , we don't do list by rootResource in informer so adding indexfield can be optional.